### DISCUSSION TITLE
Extract visitor information from web server log?

### DISCUSSION THREAD

**Discussion by mthuurne** [plausible/analytics#210] [Anything Else]
Instead of inserting a JavaScript in every page to collect information, would it be an option to analyze the web server log instead? Most of the information I see in the live demo, such as URLs requested, referrer and user agent, is found in the HTTP headers. So with the right configuration, this information can be logged by the server. In fact, the default log configuration of Apache on my PC already contains URL, referrer and user agent.

Some information such as the screen size would be hard or impossible to gather without JavaScript. But while this is useful to know, the OS distribution will already give you an idea about the ratio of desktop/laptop vs tablet/phone users.

While it's awesome that your collection script is tiny, not loading any script would be even better, in my opinion. And it would allow for analytics of user agents that don't execute JavaScript, such as web crawlers, feed readers or clients that talk to your site's API.

To be clear: I'm not advocating replacing the collection script for everyone, but it would be nice to give the site owner an alternative way of collecting information to feed into Plausible Analytics.

**Comment by tckb** [comment]
I think someone from the Changelog team also requested this "server analytics"  module. While this sounds definitely interesting, and there are many tools for analyzing the logs. 

I think it would make sense to add an "import" data feature with some format and have different exporters export into this format.  we have https://github.com/plausible/analytics/issues/142 for importing GA data.

**Comment by dch** [comment]
I would happily contribute a plugin for Elixir that does the `POST /api/event` server-side if the format is specified, and the POST can be done in a way where any API key is part of the headers or query parameters, not the body, and therefore we can safely log events locally, without needing to strip the API key.

https://docs.plausible.io/custom-domain please note that ad blockers *already* do CNAME resolution and block via IP so this is not a long-term solution if plausible isn't considered "friendly".

**Reply by metmarkosaric** [reply]
we've now documented the events API https://plausible.io/docs/events-api

**Comment by jdoubleu** [comment]
See also, that Matomo offers a [feature](https://matomo.org/docs/log-analytics-tool-how-to/#differences-using-log-analytics-vs-using-javascript-client) like this where they provide a (python) script which parses common Apache/nginx web server logs and imports all requests.

**Reply by metmarkosaric** [reply]
thanks for the feedback @jdoubleu! there are no current plans for us to go in this direction as we're focused on web analytics and on helping people remove Google Analytics from their sites. people that use server logs are already using a more privacy friendly solution and it's also very difficult to make server log data useful and usable for analytics purposes as [it's full of bots](https://plausible.io/blog/server-log-analysis). if you'd like to use Plausible without any JavaScript that's already possible with our events API. see https://plausible.io/docs/events-api
